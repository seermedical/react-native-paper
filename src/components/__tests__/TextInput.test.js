import * as React from 'react';
import { StyleSheet, Text, Platform } from 'react-native';
import { fireEvent, render } from 'react-native-testing-library';
import TextInput from '../TextInput/TextInput';
import { red500 } from '../../styles/colors';

const style = StyleSheet.create({
  inputStyle: {
    color: red500,
  },
  centered: {
    textAlign: 'center',
  },
  height: {
    height: 100,
  },
  inputFontVariant: {
    fontVariant: ['small-caps'],
  },
});

const affixTextValue = '/100';
it('correctly renders left-side icon adornment, and right-side affix adornment', () => {
  const { getByText, getByTestId, toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      onChangeText={(text) => this.setState({ text })}
      left={
        <TextInput.Icon
          name="heart"
          onPress={() => {
            console.log('!@# press left');
          }}
        />
      }
      right={
        <TextInput.Affix text={affixTextValue} textStyle={style.inputStyle} />
      }
    />
  );
  expect(() => getByText(affixTextValue)).not.toThrow();
  expect(() => getByTestId('left-icon-adornment')).not.toThrow();
  expect(() => getByTestId('right-affix-adornment')).not.toThrow();
  expect(toJSON()).toMatchSnapshot();
});

it('correctly renders left-side icon adornment, and right-side affix adornment ', () => {
  const { getByText, getByTestId, toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      onChangeText={(text) => this.setState({ text })}
      left={
        <TextInput.Affix text={affixTextValue} textStyle={style.inputStyle} />
      }
      right={
        <TextInput.Icon
          name="heart"
          onPress={() => {
            console.log('!@# press left');
          }}
        />
      }
    />
  );
  expect(() => getByText(affixTextValue)).not.toThrow();
  expect(() => getByTestId('right-icon-adornment')).not.toThrow();
  expect(() => getByTestId('left-affix-adornment')).not.toThrow();
  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies default textAlign based on default RTL', () => {
  const { toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies textAlign center', () => {
  const { toJSON } = render(
    <TextInput
      label="Flat input"
      placeholder="Type something"
      value={'Some test value'}
      style={style.centered}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies height to multiline Outline TextInput', () => {
  const { toJSON } = render(
    <TextInput
      mode="outlined"
      label="Outline Input"
      placeholder="Type Something"
      value={'Some test value'}
      style={style.height}
      multiline
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('correctly applies error state Outline TextInput', () => {
  const { getByTestId } = render(
    <TextInput
      mode="outlined"
      label="Outline Input with error"
      placeholder="Type Something"
      value={'Some test value'}
      error
    />
  );

  const outline = getByTestId('text-input-outline');
  expect(outline.props.style).toEqual(
    expect.arrayContaining([expect.objectContaining({ borderWidth: 1 })])
  );
});

it('correctly applies focused state Outline TextInput', () => {
  const { getByTestId } = render(
    <TextInput
      mode="outlined"
      label="Outline Input with error"
      placeholder="Type Something"
      value={'Some test value'}
      error
    />
  );

  const outline = getByTestId('text-input-outline');
  expect(outline.props.style).toEqual(
    expect.arrayContaining([expect.objectContaining({ borderWidth: 1 })])
  );

  fireEvent(getByTestId('text-input-outlined'), 'focus');

  expect(outline.props.style).toEqual(
    expect.arrayContaining([expect.objectContaining({ borderWidth: 2 })])
  );
});

it('contains patch spacing for flat input when ios and multiline', () => {
  Platform.OS = 'ios';
  const { getByTestId } = render(
    <TextInput
      label="Flat input"
      multiline
      placeholder="Type something"
      value={'Some test value'}
      onChangeText={(text) => this.setState({ text })}
    />
  );
  expect(() => getByTestId('patch-container')).not.toThrow();
});

it('correctly applies a component as the text label', () => {
  const { toJSON } = render(
    <TextInput
      label={<Text style={style.inputStyle}>Flat input</Text>}
      placeholder="Type something"
      value={'Some test value'}
    />
  );

  expect(toJSON()).toMatchSnapshot();
});

it('renders flat input with custom font variant', () => {
  const { getByTestId } = render(<TextInput style={style.inputFontVariant} />);

  expect(getByTestId('text-input-flat').props.style).toEqual(
    expect.arrayContaining([
      expect.objectContaining({ fontVariant: ['small-caps'] }),
    ])
  );
});

it('renders outlined input with custom font variant', () => {
  const { getByTestId } = render(
    <TextInput mode="outlined" style={style.inputFontVariant} />
  );

  expect(getByTestId('text-input-outlined').props.style).toEqual(
    expect.arrayContaining([
      expect.objectContaining({ fontVariant: ['small-caps'] }),
    ])
  );
});

it('renders input placeholder initially with an empty space character', () => {
  const { getByTestId } = render(
    <TextInput multiline label="Multiline input" />
  );

  expect(getByTestId('text-input-flat').props.placeholder).toBe(' ');
});

describe('maxFontSizeMultiplier', () => {
  const createInput = (type, maxFontSizeMultiplier) => {
    return (
      <TextInput mode={type} maxFontSizeMultiplier={maxFontSizeMultiplier} />
    );
  };

  it('should have default value in flat input', () => {
    const { getByTestId } = render(createInput('flat'));

    expect(getByTestId('text-input-flat').props.maxFontSizeMultiplier).toBe(
      1.5
    );
  });

  it('should have default value in outlined input', () => {
    const { getByTestId } = render(createInput('outlined'));

    expect(getByTestId('text-input-outlined').props.maxFontSizeMultiplier).toBe(
      1.5
    );
  });

  it('should have correct passed value in flat input', () => {
    const { getByTestId } = render(createInput('flat', 2));

    expect(getByTestId('text-input-flat').props.maxFontSizeMultiplier).toBe(2);
  });

  it('should have correct passed value in outlined input', () => {
    const { getByTestId } = render(createInput('outlined', 2));

    expect(getByTestId('text-input-outlined').props.maxFontSizeMultiplier).toBe(
      2
    );
  });

  it('should have passed null value in flat input', () => {
    const { getByTestId } = render(createInput('flat', null));

    expect(getByTestId('text-input-flat').props.maxFontSizeMultiplier).toBe(
      null
    );
  });

  it('should have passed null value in outlined input', () => {
    const { getByTestId } = render(createInput('outlined', null));

    expect(getByTestId('text-input-outlined').props.maxFontSizeMultiplier).toBe(
      null
    );
  });
});
