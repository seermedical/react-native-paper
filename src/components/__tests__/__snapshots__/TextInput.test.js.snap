// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly applies a component as the text label 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#f44336",
            }
          }
        >
          Flat input
        </Text>
      </Text>
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#f44336",
            }
          }
        >
          Flat input
        </Text>
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=" "
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      testID="text-input-flat"
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
</View>
`;

exports[`correctly applies default textAlign based on default RTL 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=" "
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      testID="text-input-flat"
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
</View>
`;

exports[`correctly applies height to multiline Outline TextInput 1`] = `
<View
  style={Object {}}
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 6,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "borderColor": "rgba(0, 0, 0, 0.54)",
          "borderRadius": 4,
          "borderWidth": 1,
        },
      ]
    }
    testID="text-input-outline"
  />
  <View>
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
          },
          Object {
            "minHeight": 100,
            "paddingTop": 8,
          },
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 3,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "backgroundColor": "#f6f6f6",
              "bottom": 4,
              "left": 10,
              "maxHeight": 2,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 6,
              "transform": Array [
                Object {
                  "translateX": -3,
                },
              ],
              "width": 12,
            }
          }
        />
        <Text
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "#f6f6f6",
              "color": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 18,
              "maxWidth": -28,
              "opacity": 1,
              "paddingHorizontal": 0,
              "position": "absolute",
              "textAlign": "left",
              "top": 59,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
                Object {
                  "scaleY": 0.2,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
        <Text
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 14,
              "position": "absolute",
              "textAlign": "left",
              "top": 58,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
        <Text
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingHorizontal": 14,
              "position": "absolute",
              "textAlign": "left",
              "top": 58,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": -52,
                },
                Object {
                  "scale": 0.75,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Outline Input
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=" "
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "paddingHorizontal": 14,
              "zIndex": 1,
            },
            Object {
              "height": 100,
            },
            Object {
              "paddingBottom": 24,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontVariant": undefined,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "top",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
        value="Some test value"
      />
    </View>
  </View>
</View>
`;

exports[`correctly applies textAlign center 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 12,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=" "
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          false,
          Array [
            Object {},
          ],
        ]
      }
      testID="text-input-flat"
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
</View>
`;

exports[`correctly renders left-side icon adornment, and right-side affix adornment  1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 3,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 12,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=" "
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 12,
            "paddingRight": 44,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Object {
            "marginLeft": 0,
            "marginRight": 36,
            "paddingLeft": 36,
            "paddingRight": 8,
          },
        ]
      }
      testID="text-input-flat"
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "left": 12,
        "opacity": 1,
        "position": "absolute",
        "top": null,
      }
    }
  >
    <Text
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          },
          Object {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
          },
          Object {
            "color": "#f44336",
          },
        ]
      }
    >
      /100
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "position": "absolute",
          "width": 24,
        },
        Object {
          "right": 12,
          "top": 20,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`correctly renders left-side icon adornment, and right-side affix adornment 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(231, 231, 231)",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.26)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
        "zIndex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 0,
          "paddingTop": 0,
        },
        Object {
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 11,
            },
          ],
          "zIndex": 3,
        }
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        onLayout={[Function]}
        style={
          Object {
            "color": "#6200ee",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 44,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
      <Text
        maxFontSizeMultiplier={1.5}
        numberOfLines={1}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
            "left": 0,
            "opacity": 0,
            "paddingLeft": 44,
            "paddingRight": 44,
            "position": "absolute",
            "textAlign": "left",
            "top": 34,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": -16,
              },
              Object {
                "scale": 0.75,
              },
            ],
            "writingDirection": "ltr",
          }
        }
      >
        Flat input
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      editable={true}
      maxFontSizeMultiplier={1.5}
      multiline={false}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=" "
      placeholderTextColor="rgba(0, 0, 0, 0.54)"
      rejectResponderTermination={true}
      selectionColor="#6200ee"
      style={
        Array [
          Object {
            "flexGrow": 1,
            "margin": 0,
          },
          Object {
            "paddingLeft": 44,
            "paddingRight": 44,
          },
          Object {
            "height": 64,
          },
          Object {
            "paddingBottom": 4,
            "paddingTop": 24,
          },
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 16,
            "fontVariant": undefined,
            "fontWeight": undefined,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          false,
          Object {
            "marginLeft": 36,
            "marginRight": 0,
            "paddingLeft": 8,
            "paddingRight": 36,
          },
        ]
      }
      testID="text-input-flat"
      underlineColorAndroid="transparent"
      value="Some test value"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 24,
          "justifyContent": "center",
          "position": "absolute",
          "width": 24,
        },
        Object {
          "left": 12,
          "top": 20,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "width": 36,
            },
            undefined,
            Object {
              "margin": 0,
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#000000",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 12,
        "top": null,
      }
    }
  >
    <Text
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
          },
          Object {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": undefined,
          },
          Object {
            "color": "#f44336",
          },
        ]
      }
    >
      /100
    </Text>
  </View>
</View>
`;
